runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -3917,6 +3917,53 @@ J9::CodeGenerator::collectSymRefs(
    }
 
 
+
+void
+J9::CodeGenerator::fixUpProfiledInterfaceGuardTest()
+   {
+   TR::Compilation *comp = self()->comp();
+   TR::CFG * cfg = comp->getFlowGraph();
+   TR::Block *currentBlock = NULL;
+   for (TR::TreeTop * treeTop = comp->getStartTree(); treeTop; treeTop = treeTop->getNextTreeTop())
```

Comment:
We are you walking all TreeTops - a profiled guard has to be at the end of a basic block so you can walk the blocks in treetop order checking their exits which will save time...

Reply:
Consider using one of the Block iterators available in ILWalk.cpp to accomplish this walk.

Reply:
Added change to walk blocks in CFG and checking Exits (https://github.com/eclipse/openj9/pull/4721/commits/36292aa9f2fcefeedc9fe76a5f0929a1ce9c2d68)